src/main/java/org/assertj/vavr/internal/Maps.java

```diff
@@ -0,0 +1,135 @@
+     *
+     * @param <K>     key type
+     * @param <V>     value type
+     * @param info    contains information about the assertion.
+     * @param actual  the given {@code Map}.
+     * @param entries the entries that are expected to be in the given {@code Map}.
+     * @throws NullPointerException     if the array of entries is {@code null}.
+     * @throws IllegalArgumentException if the array of entries is empty.
+     * @throws NullPointerException     if any of the entries in the given array is {@code null}.
+     * @throws AssertionError           if the given {@code Map} is {@code null}.
+     * @throws AssertionError           if the given {@code Map} does not contain the given entries.
+     */
+    public <K, V> void assertContains(AssertionInfo info, Map<K, V> actual,
+                                      Tuple2<K, V>[] entries) {
+        failIfNull(entries);
+        assertNotNull(info, actual);
+        // if both actual and values are empty, then assertion passes.
+        if (actual.isEmpty() && entries.length == 0) return;
+        failIfEmptySinceActualIsNotEmpty(entries);
+        final Set<Tuple2<? extends K, ? extends V>> notFound = Array
+                .of(entries)
+                .foldLeft(HashSet.empty(), (set, tuple) -> {
+                    if (actual.contains(tuple)) {
+                        return set;
+                    } else {
+                        return set.add(tuple);
+                    }
+                });
+        if (!notFound.isEmpty()) {
+            throw failures.failure(info, shouldContain(actual, entries, notFound));
+        }
+    }
+
+    /**
+     * Asserts that the given {@code Map} does not contain the given entries.
+     *
+     * @param <K>     key type
+     * @param <V>     value type
+     * @param info    contains information about the assertion.
+     * @param actual  the given {@code Map}.
+     * @param entries the entries that are expected to be in the given {@code Map}.
+     * @throws NullPointerException     if the array of entries is {@code null}.
+     * @throws IllegalArgumentException if the array of entries is empty.
+     * @throws NullPointerException     if any of the entries in the given array is {@code null}.
+     * @throws AssertionError           if the given {@code Map} is {@code null}.
+     * @throws AssertionError           if the given {@code Map} contains any of the given entries.
+     */
+    public <K, V> void assertDoesNotContain(AssertionInfo info, Map<K, V> actual,
+                                            Tuple2<K, V>[] entries) {
+        failIfNullOrEmpty(entries);
+        assertNotNull(info, actual);
+        failIfEmptySinceActualIsNotEmpty(entries);
+        final Set<Tuple2<? extends K, ? extends V>> found = Array
+                .of(entries)
+                .foldLeft(HashSet.empty(), (set, tuple) -> {
+                    if (actual.contains(tuple)) {
```

Comment:
If actual contains tuple then add it to set and return? 🤔

Reply:
That's the folding function. For a single run of it, we check whether `actual` contains a given `tuple`. If so, then we're collecting it and returning accumulator from the function.

Reply:
Mixed two different collections 🤕

Reply:
Yup, easy to do that with lambdas